### PR TITLE
Add package dependency

### PR DESCRIPTION
**Pull request by rgrinberg**
Implement #621. @diml does this look right? I think I just need to add some better validation of the package name itself.

### CODE REVIEW

**Comment by diml** [comment]
Yep. What validation are you thinking about?

**Comment by rgrinberg** [comment]
Just to verify that the package exists in `sctx.packages`.

**Comment by diml** [comment]
Ah, indeed. Maybe we can have `Package of Package.t` then, and use `Scope_info.resolve`? Then we'll get the same error messages as for `(package xxx)`.

File: src/install_rules.ml [hunk 1]
```diff
@@ -244,6 +244,9 @@ module Gen(P : Install_params) = struct
         (Utils.install_file ~package ~findlib_toolchain:ctx.findlib_toolchain)
     in
     let entries = local_install_rules entries ~package in
+    SC.add_alias_deps sctx
+      (Alias.package_install ~context:ctx ~pkg:package)
+      (List.map entries ~f:(fun (e : Install.Entry.t) -> e.src));
```
Comment (rgrinberg) [inline src/install_rules.ml]: Shouldn't we be using the destination of the symlink however? Otherwise, the symlink will not be made.

**Comment by rgrinberg** [comment]
> Ah, indeed. Maybe we can have Package of Package.t then, and use Scope_info.resolve? Then we'll get the same error messages as for (package xxx).

So I suppose we cannot depend on a package that's outside our scope? I guess that's reasonable.

File: src/install_rules.ml (hunk 1 above)
Comment (diml) [reply src/install_rules.ml]: That's what we are doing. `local_install_rules` replaces all the `src` fields by the symlinks

**Comment by diml** [comment]
> So I suppose we cannot depend on a package that's outside our scope? I guess that's reasonable.

Yes, although there is indeed an issue here. When we create a big workspace, it's likely that we also need to include the dependent packages.

**Comment by diml** [comment]
I added a test with two dependent packages and a call to ocamlfind and it's now working as expected

**Comment by rgrinberg** [comment]
What about the approach where you were saving the package along with each rule, why didn't that work? Having a global map seems of packages to files seems a little uglier.

(I tried asking this before, but I think github ate my question. Apologies if you received this through email)

**Comment by diml** [comment]
I was saving it on the symlink rule. That didn't work because the dependencies during the build are most of the time against files in `_build/default`.

**Pull request merged**